mirdata/msd.py

```diff
@@ -0,0 +1,373 @@
+    This function prints a helper message to set up an EC2 instance
+    with the MSD mounted
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    """
+
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    info_message = """
+There are several ways of downloading the Million Song Dataset,
+listed on http://millionsongdataset.com/pages/getting-dataset/.
+
+The easiest way is to create an instance on Amazon Web Services (AWS),
+and do the analysis there. First sign up https://aws.amazon.com/
+(requires entering credit card details).
+
+Then, in a on your local machine:
+
+# Install the AWS command line interface
+$ pip install awscli
+
+# Configure AWS
+#   - Create a new access key on https://console.aws.amazon.com/iam/home?region=us-east-1#/security_credentials
+#   - Enter the access key ID and secret access key
+#   - Set zone to us-east-1
+aws configure  # zone=us-east-1 and credentials
+
+# Create a new SSH key pair
+$ KEY_NAME=my-aws-keypair
+$ PEM_PATH=~/.ssh/aws-key-pair.pem
+$ aws ec2 create-key-pair --key-name $KEY_NAME --query 'KeyMaterial' --output text > $PEM_PATH
+
+# Create a Ubuntu EC2 instance
+$ UBUNTU_1804=ami-04b9e92b5572fa0d1
+$ aws ec2 run-instances --image-id $UBUNTU_1804 --count 1 --key-pair $KEY_NAME --instance-type m1.small > /tmp/instance.json
+
+# Create a volume from the MSD snapshot
+$ AVAILABILITY_ZONE=$(jq -r '.Instances[0].Placement.AvailabilityZone' /tmp/instance.json)
+$ MSD_SNAPSHOT=snap-5178cf30
+$ aws ec2 create-volume --availability-zone $AVAILABILITY_ZONE --snapshot-id $MSD_SNAPSHOT --volume-type gp2 > /tmp/volume.json
+
+# Attach the volume to the instance
+$ VOLUME_ID=$(jq -r '.VolumeId' /tmp/volume.json)
+$ INSTANCE_ID=$(jq -r '.Instances[0].InstanceId' /tmp/instance.json)
+$ DEVICE=/dev/xvdk
+$ aws ec2 attach-volume --volume-id $VOLUME_ID --instance-id $INSTANCE_ID --device $DEVICE
+
+# Connect to the instance
+$ INSTANCE_HOSTNAME=$(aws ec2 describe-instances --instance-ids $INSTANCE_ID | jq -r .Reservations[0].Instances[0].PublicDnsName)
+$ MSD_FOLDER={msd_folder}
+$ ssh -A -i $PEM_PATH ubuntu@$INSTANCE_HOSTNAME
+
+# On the instance, mount the MSD:
+$ sudo mkdir $MSD_FOLDER
+$ sudo mount $DEVICE $MSD_FOLDER
+$ sudo chmod 777 $MSD_FOLDER"
+
+""".format(
+        msd_folder=data_home
+    )
+
+    download_utils.downloader(data_home, info_message=info_message)
+
+
+def cite():
+    """Print the reference"""
+
+    cite_data = """
+=========== MLA ===========
+Thierry Bertin-Mahieux, Daniel P.W. Ellis, Brian Whitman, and Paul Lamere.
+The Million Song Dataset. In Proceedings of the 12th International Society
+for Music Information Retrieval Conference (ISMIR 2011), 2011.
+========== Bibtex ==========
+@inproceedings{Bertin-Mahieux2011,
+  author = {Thierry Bertin-Mahieux and Daniel P.W. Ellis and Brian Whitman and Paul Lamere},
```

Comment:
I think the names have to be inverted - like
`author = {Bertin-Mahieux, Thierry and ... `